sklearn/covariance/_graph_lasso.py

```diff
@@ -315,6 +210,150 @@ def graphical_lasso(
             return covariance_, precision_
 
 
+def alpha_max(emp_cov):
+    """Find the maximum alpha for which there are some non-zeros off-diagonal.
+
+    Parameters
+    ----------
+    emp_cov : ndarray of shape (n_features, n_features)
+        The sample covariance matrix.
+
+    Notes
+    -----
+    This results from the bound for the all the Lasso that are solved
+    in GraphicalLasso: each time, the row of cov corresponds to Xy. As the
+    bound for alpha is given by `max(abs(Xy))`, the result follows.
+    """
+    A = np.copy(emp_cov)
+    A.flat[:: A.shape[0] + 1] = 0
+    return np.max(np.abs(A))
+
+
+# The g-lasso algorithm
+def graphical_lasso(
+    emp_cov,
+    alpha,
+    *,
+    cov_init=None,
```

Comment:
Since the parameter `cov_init` and `eps` are not exposed by `GraphicalLasso` then we need to expose them.
We need to update the documentation of `GraphicalLasso` as well.